### DISCUSSION TITLE
How to package LoVR code for Oculus Quest (2)

### DISCUSSION THREAD

**Discussion by AlexMikhalev** [bjornbytes/lovr#510] [Q&A] [answered]
Hello Bjorn and all,
Amazing project, I wish it would be known wider!
I am currently evaluating to build simple UI in VR  - may be using Voronoi diagram or L-Systems to generate visuals, or import few simple models, nothing complex.
What is the process to package LoVR for Oculus Store? 
I have seen https://lovr.org/docs/Getting_Started_(Quest) and I have Quest 2 for testing, can someone point me to discussion/howto?

**Comment by bjornbytes** [comment, marked as answer]
Instead of pushing files to the sdcard directory to iterate, it's possible to make an apk that includes the project files.  When building an apk (see the [Compiling guide](https://lovr.org/docs/Compiling)) there is an option `CONFIG_ANDROID_ASSETS` that can be used to automatically include a lovr project in the apk.  It's done this way at build time because an apk has to be re-signed whenever its files change.

All this compile option does is include the project under the `assets` path of the apk (e.g. `assets/main.lua`, APKs are just zip archives).  It might be possible to achieve the same thing without recompiling using a tool like `apktool`, but I haven't tried it.

At least 2 people have started to go through the Oculus Store review process with lovr projects, and there have already been some small fixes made to lovr in response to problems Oculus has pointed out.  The list of requirements is public: [here](https://developer.oculus.com/resources/publish-quest-req/)

**Reply by bjornbytes** [reply]
I opened [this issue](https://github.com/bjornbytes/lovr-docs/issues/68) to improve things since I think the Distribution guide could provide more info on how to create an apk for distribution.

**Reply by jtoy** [reply]
I'm interested in learning about this as well. I'll play around with this myself too.
